py/server/deephaven/time.py

```diff
@@ -155,1885 +135,485 @@ def time_zone_alias_rm(alias: str) -> bool:
+        dt (Union[None, Duration]): A Deephaven duration.  If None is provided, None is returned.
 
     Returns:
-        Instant or None if the input is NULL_LONG.
-
-    Raises:
-        DHError
+        datetime.timedelta
     """
     try:
-        return _JDateTimeUtils.epochMicrosToInstant(micros)
+        if dt is None:
+            return None
+        elif isinstance(dt, Duration.j_type):
+            return datetime.timedelta(seconds=dt.getSeconds(), microseconds=dt.getNano() // 1000)
+        elif isinstance(dt, Period.j_type):
+            y = dt.getYears()
+            m = dt.getMonths()
+            d = dt.getDays()
+
+            if y or m:
+                raise Exception("Unsupported conversion: " + str(type(dt)) +
+                                " -> datetime.timedelta: Periods must only be days or weeks")
+
+            return datetime.timedelta(days=d)
+        else:
+            raise Exception("Unsupported conversion: " + str(type(dt)) + " -> datetime.timedelta")
     except Exception as e:
         raise DHError(e) from e
 
 
-def epoch_millis_to_instant(millis: int) -> Instant:
-    """ Converts milliseconds from the Epoch to an Instant.
+def to_timedelta64(dt: Union[None, Duration, Period]) -> Optional[np.timedelta64]:
+    """
+    Converts a Deephaven duration or period to a numpy timedelta64.
 
     Args:
-        millis (int): Milliseconds since Epoch.
+        dt (Union[None, Duration, Period]): A Deephaven duration or period.  If None is provided, None is returned.
 
     Returns:
-        Instant or None if the input is NULL_LONG.
+        np.timedelta64
 
     Raises:
         DHError
     """
     try:
-        return _JDateTimeUtils.epochMillisToInstant(millis)
+        if dt is None:
+            return None
+        elif isinstance(dt, Duration.j_type):
+            return np.timedelta64(dt.toNanos(), 'ns')
+        elif isinstance(dt, Period.j_type):
+            d = dt.getDays()
+            m = dt.getMonths()
+            y = dt.getYears()
+
+            count = (1 if d else 0) + (1 if m else 0) + (1 if y else 0)
+
+            if count == 0:
+                return np.timedelta64(0, 'D')
+            elif count > 1:
+                raise Exception("Unsupported conversion: " + str(type(dt)) +
+                                " -> datetime.timedelta64: Periods must be days, months, or years")
+            elif y:
+                return np.timedelta64(y, 'Y')
+            elif m:
+                return np.timedelta64(m, 'M')
+            elif d:
+                return np.timedelta64(d, 'D')
+            else:
+                raise Exception("Unsupported conversion: " + str(type(dt)) + " -> datetime.timedelta64: (" + dt + ")")
+        else:
+            raise Exception("Unsupported conversion: " + str(type(dt)) + " -> datetime.timedelta64")
     except Exception as e:
         raise DHError(e) from e
 
 
-def epoch_seconds_to_instant(seconds: int) -> Instant:
```

Comment:
Why is all this stuff going away? Does the new API somehow capture all of it? Is it no longer useful?